app/models/fire_cloud_client.rb

```diff
@@ -1144,19 +1144,39 @@ def get_workspace_file(workspace_namespace, workspace_name, filename)
 
 	# add a file to a workspace bucket
 	#
-  # * *params*
+	# * *params*
 	#   - +workspace_namespace+ (String) => namespace of workspace
 	#   - +workspace_name+ (String) => name of workspace
 	#   - +filepath+ (String) => path to file
 	#   - +filename+ (String) => name of file
 	#   - +opts+ (Hash) => extra options for create_file, see
-  #     https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-storage/v0.23.2/google/cloud/storage/bucket?method=create_file-instance
+	#     https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-storage/v0.23.2/google/cloud/storage/bucket?method=create_file-instance
 	#
-  # * *return*
-  #   - +Google::Cloud::Storage::File+
+	# * *return*
+	#   - +Google::Cloud::Storage::File+
 	def create_workspace_file(workspace_namespace, workspace_name, filepath, filename, opts={})
 		bucket = self.get_workspace_bucket(workspace_namespace, workspace_name)
-		bucket.create_file filepath, filename, opts
+		first_two_bytes = File.open(filepath).read(2)
+		gzip_signature = "\x1F\x8B".force_encoding(Encoding::ASCII_8BIT) # per IETF
+		file_is_gzipped = (first_two_bytes == gzip_signature)
+		if file_is_gzipped or filepath.last(4) == '.bam' or filepath.last(5) == '.cram'
+			Rails.logger.info "Direct upload"
```

Comment:
I like to add timestamps on the logger calls with "#{Time.now}: " and the message.  Since this gets called as a part of study.send_to_firecloud, I don't think it needs more info than that.